route_guide/route_guide_client.cc

```diff
@@ -23,238 +23,324 @@
-                      << feature->location().longitude()/kCoordFactor_ << std::endl;
-          } else {
-            call->Finish();
-          }
-        },
-        [&](auto*, const Status& status) {
-          done.Notify(status);
-        });
+ public:
+  // The goal of GetFeature is just to implement async unary rpc call.
+  // Client sends to the server the request which contains Point's coordi-
+  // nates (longitude, latitude). The server checks received Point's coordi-
+  // nates, if database has the specific address which corresponds to Point's
+  // coordinates - the server will response with it address. GetFeature
+  // returns lambda.
+  auto GetFeature(long latitude, long longitude) {
+    // The idea of Data structure is just to use it in the .context() callback
+    // in order to have the way of injecting some data on a stack, or as
+    // a part of the eventual data structure and manipulate with this data in
+    // multiple callbacks (.start(), .fail(), .ready(), .body(), etc).
+    struct Data {
+      Point point;      // Request from the client
+      Feature feature;  // Response from the server
+    };
 
-    Status status = done.Wait();
-    if (status.ok()) {
-      std::cout << "ListFeatures rpc succeeded." << std::endl;
-    } else {
-      std::cout << "ListFeatures rpc failed." << std::endl;
-    }
+    return [latitude, longitude, this]() {
+      // Call() function gets the rpc name and the host name as parameters
+      // and returns Composable structure which is composed with Handler()
+      // call using overloaded operator "|" in order to handle calls from
+      // server. Actually Handler() returns Composable structure too, so it
+      // can be composed with other tasks. (You can check Composable
+      // structure's implementation in the stout/grpc directory in handler.h)
+      return client_->Call<RouteGuide, Point, Feature>("GetFeature") |
+             (Client::Handler()
+                  // Passing Data structure to the context
+                  .context(Data{MakePoint(latitude, longitude), Feature{}})
+                  // Send the request with Point's data to the server. First
+                  // argument in the .ready() callback gets the context. The
+                  // second argument gets the call. You can send requests to
+                  // the server using Write*() family functions. In this
+                  // case we just send a unique request and finish our call
+                  // with WriteLast function.
+                  .ready([](auto&& data, auto& call) {
+                    call.WriteLast(data.point);
+                  })
+                  // .body() callback will be invoked every time the server
+                  // will send the response. First argument in lambda gets
+                  // the context, second - the call, third - the response
+                  // object from the server.
```

Comment:
Some English suggestions:
```suggestion
                  // The `.body()` callback is be invoked every time the server
                  // sends a response. The lambda's first argument gets the 
                  // context. The second gets the call. The third gets the 
                  // response object from the server.
```